doc/man3/OSSL_HPKE_CTX_new.pod

```diff
@@ -0,0 +1,539 @@
+OSSL_HPKE_encap().
+
+OSSL_HPKE_CTX_set1_authpub() can be used by the receiver to set the senders
+encoded pub key I<pub> of size I<publen> into the B<OSSL_HPKE_CTX> I<ctx> before
+calling OSSL_HPKE_decap().
+For EC curves, I<pub> is in an uncompressed encoding format.
+
+=head2 Pre-Shared Key HPKE modes
+
+HPKE also defines a symmetric equivalent to the authentication described above
+using a pre-shared key (PSK) and a PSK identifier. PSKs can be used with the
+B<OSSL_HPKE_MODE_PSK> and B<OSSL_HPKE_MODE_PSKAUTH> modes.
+
+OSSL_HPKE_CTX_set1_psk() sets the PSK identifier I<pskid> string, and PSK buffer
+I<psk> of size I<psklen> into the I<ctx>. If required this must be called
+before OSSL_HPKE_encap() or OSSL_HPKE_decap().
+As per RFC9180, if required, both I<psk> and I<pskid> must be set to non NULL values.
+As PSKs are symmetric the same calls must happen on both sender and receiver
+sides.
+
+=head2 Deterministic key generation for senders
+
+Normally the senders ephemeral private key is generated randomly inside
+OSSL_HPKE_encap() and remains secret.
+OSSL_HPKE_CTX_set1_ikme() allows the user to override this behaviour by
+setting a deterministic input key material I<ikm> of size I<ikmlen> into
+the B<OSSL_HPKE_CTX> I<ctx>.
+If required OSSL_HPKE_CTX_set1_ikme() can optionally be called before
+OSSL_HPKE_encap().
+I<ikmlen> should be greater than or equal to OSSL_HPKE_get_recommended_ikmelen().
+
+It is generally undesirable to use OSSL_HPKE_CTX_set1_ikme(), since it
+exposes the relevant secret to the application rather then preserving it
+within the library, and is more likely to result in use of predictable values
+or values that leak.
+
+=head2 Re-sequencing
+
+Some protocols may have to deal with packet loss while still being able to
+decrypt arriving packets later. We provide a way to set the increment used for
+the nonce to the next subsequent call to OSSL_HPKE_seal() or OSSL_HPKE_open().
+The OSSL_HPLE_CTX_set_seq() API can be used for such purposes with the I<seq>
+parameter value resetting the internal nonce to be used for the next call.
+
+A baseline nonce value is established based on the encapsulation or
+decapsulation operation and is then incremented by 1 for each call to seal or
+open. (In other words, the I<seq> is a zero-based counter.)
+
+If a caller needs to determine how many calls to seal or open have been made
+the OSSL_HPKE_CTX_get_seq() API can be used to retrieve the increment (in the
+I<seq> output) that will be used in the next call to seal or open. That would
+return 0 before the first call a sender made to OSSL_HPKE_seal() and 1 after
+that first call.
+
+For compatibility with other implementations these I<seq> increments are
+represented as a I<uint64_t>.
+
+Note that re-use of the same nonce and key with different plaintexts is very
+dangerous and can lead to loss of confidentiality. Applications therefore need
+to exercise extreme caution in using these APIs and would be better off avoiding
+them entirely.
```

Comment:
Maybe we should allow that only with _open() call not _seal()? Otherwise this is really dangerous. If the application needs resending the same sealed data it can cache the sealed data instead of re-encrypting?

Reply:
I hadn't considered making this only available for ``_open()`` - I'll ponder that a bit whilst doing the compressed form test mentioned above, and get back, but welcome other inputs.

Reply:
Having thought about this some, I think yes, only allowing ``_set_seq()`` to have an effect for subsequent ``_open()`` operations is doable and safer. I plan to do that tomorrow (time permitting, might be Friday). Would still appreciate more input on that meantime.  That'll also mean another small chunk of test code, which is fine.
Thanks for the excellent comment!

Reply:
If you want to do it in a subsequent PR, that would be also fine.

Reply:
Actually, yeah that's a better plan for 3 reasons: 1) I'm still not sure if there might be a use case that needs to set an initial sequence on the sending side (before ``_encap()``) but mostly 2) adding that restriction may create some subtle constraints on the ordering of API calls, and of course 3) nonce-reuse is tricky and warrants going slowly;-) So if it's ok, yes, I'd prefer land this PR first and then immediately create another to improve safety and reduce the potential for nonce-reuse.